controllers/external/util.go

```diff
@@ -49,9 +49,31 @@ func Get(ctx context.Context, c client.Client, ref *corev1.ObjectReference, name
 	return obj, nil
 }
 
+type CloneTemplateInput struct {
+	// TemplateRef is a reference to the template that needs to be cloned.
+	// +required
+	TemplateRef *corev1.ObjectReference
+
+	// Namespace is the Kuberentes namespace the cloned object should be created into.
+	// +required
+	Namespace string
+
+	// ClusterName is the cluster this object is linked to.
+	// +required
+	ClusterName string
+
+	// OwnerRef is an optional OwnerReference to attach to the cloned object.
+	// +optional
+	OwnerRef *metav1.OwnerReference
+
+	// Labels is an optional map of labels to be added to the object.
+	// +optional
+	Labels map[string]string
+}
+
 // CloneTemplate uses the client and the reference to create a new object from the template.
-func CloneTemplate(ctx context.Context, c client.Client, ref *corev1.ObjectReference, namespace, clusterName string, owner *metav1.OwnerReference) (*corev1.ObjectReference, error) {
-	from, err := Get(ctx, c, ref, namespace)
+func CloneTemplate(ctx context.Context, c client.Client, in *CloneTemplateInput) (*corev1.ObjectReference, error) {
```

Comment:
Should we put the client in the input struct?

Reply:
I initially had it in there, but it felt weird because it was the only field that didn't describe the end object, if that makes sense. I can re-add it, no strong preferences

Reply:
The purist in me says it's either `(context, input)` or `(lots, and, lots, of, args)` 😄

Reply:
Fixed